source/backend/discovery/src/lib/awsClient.js

```diff
@@ -573,10 +576,27 @@ function createCognitoClient(credentials, region) {
 
 }
 
+function createDynamoDBStreamsClient(credentials, region) {
+    const dynamoDBStreamsClient = new DynamoDBStreams({customUserAgent, region, credentials});
+    // this API only has a TPS of 10 so we set it artificially low to avoid rate limiting
+    const describeStreamThrottler = pThrottle({
+        limit: 8,
+        interval: 1000
+    });
+
+    return {
+        async describeStream(streamArn) {
+            const describeStreamInfo = describeStreamThrottler(async (streamArn) => await dynamoDBStreamsClient.describeStream({StreamArn: streamArn}));
```

Comment:
This `await` needs to be at the top level, not in the in the throttler. Also, we can save having to name this variable by using destructuring:

```js
const {StreamDescription} = await describeStreamThrottler(async (streamArn) => dynamoDBStreamsClient.describeStream({StreamArn: streamArn}));
return StreamDescription;
```

Reply:
It has no effect when it's at the top level according to my IDE. Going into the throttler there were no complaints though.

Reply:
When I ran the code I got `undefined` because it wass trying to get `StreamDescription` off of an unwrapped promise.

Reply:
My guess is that your IDE doesn't know that `describeStreamThrottler` returns a promise.